mailet/base/src/main/java/org/apache/mailet/base/MailetUtil.java

```diff
@@ -117,21 +117,22 @@ public static int getInitParameterAsStrictlyPositiveInteger(String condition, in
     }
 
     public static int getInitParameterAsStrictlyPositiveInteger(String condition) throws MessagingException {
-        return getInitParameterAsStrictlyPositiveInteger(condition, Optional.<String>absent());
+        return getInitParameterAsStrictlyPositiveInteger(condition, Optional.empty());
     }
 
     public static int getInitParameterAsStrictlyPositiveInteger(String condition, Optional<String> defaultValue) throws MessagingException {
-        Optional<String> value = Optional.fromNullable(condition)
-            .or(defaultValue);
+        String value = Optional.ofNullable(condition)
+            .filter(Strings::isNullOrEmpty)
+            .orElse(defaultValue.orElse(null));
```

Comment:
why this change?

Reply:
defaultValue is an Optional
The java Optional does not have default value as an Optional.
Anyway, I remove "filter" which should be confuse